### PR TITLE
Adding structure and enhancements to user documentation 

### PR DESCRIPTION
**Pull request by Kostov6**
**How to categorize this PR?**

/area documentation
/kind enhancement

**What this PR does / why we need it**:
This PR introduces minor content rephrasing and tries to make user content improvement based on the following heuristic:
- selecting better titles and adding descriptions where needed by trying to reference as much content keywords as possible while remaining readable . Benefit of that is that the search functionality of the website gives more weight to matched words in titles and descriptions of pages. This will aid in improving website search quality. Also the search shows the description under the title so search results will be more clear
- adding user content structure for better content navigation through the navbar sections
- adding meaningful order of pages. My approach was to make it based on content simplicity and expected user benefit from the content (simpler content and more user benefit push a given page to the top) with simplicity having more weight than expected benefit. The reason behind this is that "user benefit" has high variance (different users benefit differently from different content) and is more subjective than content simplicity which is more objective. This ordering heuristic makes the order more stable while pushing more beneficial pages to the top
- fixing broken links. Even if some links were missed out https://github.com/gardener/documentation/pull/508 will introduce aliases that will redirect the old links to the new links. Even if that fails we have a 404 page that handles moved pages (ex. https://gardener.cloud/foo/gardenlet/ )
- adhering to content locality: similar pages/sections are near each other. Example are sections `Configuring the Shoot Spec`, `Containerd Configurations`, `DNS Configurations`

Also some pages are moved from user to other persona(ex `ipv6.md`) and some pages are divided (ex `default_seccom_profile_seed.md`)

**Which issue(s) this PR fixes**:
Part of https://github.com/gardener/documentation/issues/488

**Special notes for your reviewer**:
I tried to adhere to the heuristic listed above as much as possible and to construct it to have as much value as possible. Still those kind of changes are highly subjective and I don't have any strong opinions so I'm open to any request for change. Please when requesting change provide a reason based on:
- enhancement of the heuristic the change was based upon
- pointing out where a change does not reflect the PR/proposed heuristic

How the section will look

Screenshot 2024-06-27 at 17 35 34

And `Configuring the Shoot Spec` 

Screenshot 2024-06-27 at 17 36 38

And it's section with the descriptions

Screenshot 2024-06-27 at 17 39 51

**Release note**:

```doc user
Majority of user content is moved to a dedicated section
```

### CODE REVIEW

File: docs/usage/containerd/_index.md [hunk 1]
```diff
@@ -0,0 +1,4 @@
+---
+title: Containerd Configurations
```
Comment (n-boshnakov) [inline docs/usage/containerd/_index.md]:
```suggestion
title: containerd Configurations
```

If the title is stylized all lowercase, then it should be `containerd` instead of `Containerd`

File: docs/usage/nessesary_considerations/_index.md [hunk 2]
```diff
@@ -0,0 +1,5 @@
+---
+title: Considerations for specific scenarios
```
Comment (n-boshnakov) [inline docs/usage/nessesary_considerations/_index.md]:
```suggestion
title: Considerations for Specific Scenarios
```

File: docs/usage/nessesary_considerations/csi_components.md [hunk 3]
```diff
@@ -1,5 +1,6 @@
 ---
-title: CSI Components
+title: Nessesary labeling for Custom CSI Components
```
Comment (n-boshnakov) [inline docs/usage/nessesary_considerations/csi_components.md]:
```suggestion
title: Necessary Labeling for Custom CSI Components
```

File: docs/usage/nessesary_considerations/node-readiness.md [hunk 4]
```diff
@@ -13,7 +17,7 @@ However, several proposals have been rejected because the problem can be solved
 
 Gardener makes sure that workload Pods are only scheduled to `Nodes` where all node-critical components required for running workload Pods are ready. 
 For this, Gardener follows the proposed solution by the Kubernetes community and registers new `Node` objects with the `node.gardener.cloud/critical-components-not-ready` taint (effect `NoSchedule`).
-gardener-resource-manager's [`Node` controller](../concepts/resource-manager.md#node-controller) reacts on newly created `Node` objects that have this taint.
+gardener-resource-manager's [`Node` controller](../../concepts/resource-manager.md#node-controller) reacts on newly created `Node` objects that have this taint.
```
Comment (n-boshnakov) [inline docs/usage/nessesary_considerations/node-readiness.md]:
```suggestion
`gardener-resource-manager`'s [`Node` controller](../../concepts/resource-manager.md#node-controller) reacts on newly created `Node` objects that have this taint.
```

File: docs/usage/operating_through_annotations/_index.md [hunk 5]
```diff
@@ -0,0 +1,4 @@
+---
+title: Operating via Shoot annotations
```
Comment (n-boshnakov) [inline docs/usage/operating_through_annotations/_index.md]:
```suggestion
title: Operating via Shoot Annotations
```

File: docs/usage/operating_through_annotations/shoot_operations.md [hunk 6]
```diff
@@ -1,4 +1,9 @@
-# Trigger Shoot Operations
+---
+title: Trigger Shoot Operations through annotations
```
Comment (n-boshnakov) [inline docs/usage/operating_through_annotations/shoot_operations.md]:
```suggestion
title: Trigger Shoot Operations Through Annotations
```

File: docs/usage/operating_through_annotations/shoot_operations.md [hunk 7]
```diff
@@ -1,4 +1,9 @@
...
+weight: 1
+---
+
+# Trigger Shoot Operations through annotations
```
Comment (n-boshnakov) [inline docs/usage/operating_through_annotations/shoot_operations.md]:
```suggestion
# Trigger Shoot Operations Through Annotations
```

File: docs/usage/shoot_basics/basic-configuration.md [hunk 8]
```diff
@@ -0,0 +1,6 @@
+---
+title: Basic Shoot configurations
```
Comment (n-boshnakov) [inline docs/usage/shoot_basics/basic-configuration.md]:
```suggestion
title: Basic Shoot Configurations
```

File: docs/usage/shoot_basics/creating-shoot.md [hunk 9]
```diff
@@ -0,0 +1,6 @@
+---
+title: Creating a Shoot cluster
```
Comment (n-boshnakov) [inline docs/usage/shoot_basics/creating-shoot.md]:
```suggestion
title: Creating a Shoot Cluster
```

File: docs/usage/shoot_basics/shoot_status.md [hunk 10]
```diff
@@ -1,10 +1,15 @@
+---
+weight: 8
+description: Shoot conditions, constraints and error codes
```
Comment (n-boshnakov) [inline docs/usage/shoot_basics/shoot_status.md]:
```suggestion
description: Shoot conditions, constraints, and error codes
```

File: docs/usage/shoot_basics/shoot_status.md [hunk 11]
```diff
@@ -1,10 +1,15 @@
...
+
 # Shoot Status
 
-This document provides an overview of the [ShootStatus](../api-reference/core.md#shootstatus).
+This document provides an overview of the [ShootStatus](../../api-reference/core.md#shootstatus).
 
 ## Conditions
 
-The Shoot status consists of a set of conditions. A [Condition](../api-reference/core.md#condition) has the following fields:
+The Shoot status consists of a set of conditions. They represent the latest available observations of a Shoots’s current state. A [Condition](../../api-reference/core.md#condition) has the following fields:
```
Comment (n-boshnakov) [inline docs/usage/shoot_basics/shoot_status.md]:
```suggestion
The Shoot status consists of a set of conditions. They represent the latest available observations of a Shoot’s current state. A [Condition](../../api-reference/core.md#condition) has the following fields:
```

File: docs/usage/shoot_high_availability_best_practices.md [hunk 12]
```diff
@@ -1,3 +1,7 @@
+---
+title: High avaliability best practices
```
Comment (n-boshnakov) [inline docs/usage/shoot_high_availability_best_practices.md]:
```suggestion
title: High Avaliability Best Practices
```

File: docs/usage/shoot_settings/shoot_autoscaling.md [hunk 13]
```diff
@@ -17,21 +22,21 @@ Consequently, please refer to the [official documentation](https://github.com/ku
...
 > Only some `cluster-autoscaler` flags can be configured per worker pool, and is limited by NodeGroupAutoscalingOptions of the upstream community Kubernetes repository. This list can be found [here](https://github.com/gardener/autoscaler/blob/machine-controller-manager-provider/cluster-autoscaler/config/autoscaling_options.go#L37-L55).
 
 ## Vertical Pod Auto-Scaling
 
 This form of auto-scaling is not enabled by default and must be explicitly enabled in the `Shoot` by setting `.spec.kubernetes.verticalPodAutoscaler.enabled=true`.
-The reason is that it was only introduced lately, and some end-users might have already deployed their own VPA into their clusters, i.e., enabling it by default would interfere with such custom deployments and lead to issues, eventually.
+The reason is that it was only introduced lately, and some end-users might have already deployed their own VPA into their clusters, i.e., enabling it by default would interfere with such custom deployments and lead to issues, eventually. Also if `ShootVPAEnabledByDefault` admissionPlugins is enabled then `.spec.kubernetes.verticalPodAutoscaler.enabled` will be set to true.
```
Comment (n-boshnakov) [inline docs/usage/shoot_settings/shoot_autoscaling.md]:
```suggestion
The reason is that it was only introduced lately, and some end-users might have already deployed their own VPA into their clusters, i.e., enabling it by default would interfere with such custom deployments and lead to issues, eventually. Also, if `ShootVPAEnabledByDefault` admissionPlugins is enabled, then `.spec.kubernetes.verticalPodAutoscaler.enabled` will be set to true.
```

File: docs/usage/shoot_settings/shoot_supported_architectures.md [hunk 14]
```diff
@@ -0,0 +1,53 @@
+---
+weight: 5
+title: Shoot Workers
+description: How to specify a shoot worker group given the respective `CloudProfile` 
+---
+
+# Configuring a shoot worker group
```
Comment (n-boshnakov) [inline docs/usage/shoot_settings/shoot_supported_architectures.md]:
```suggestion
# Configuring a Shoot Worker Group 
```

File: docs/usage/shoot_updates_and_upgrades/_index.md [hunk 15]
```diff
@@ -0,0 +1,4 @@
+---
+title: Shoot updates and upgrades
```
Comment (n-boshnakov) [inline docs/usage/shoot_updates_and_upgrades/_index.md]:
```suggestion
title: Shoot Updates and Upgrades
```

**Review by n-boshnakov — CHANGES_REQUESTED** [review]

File: docs/usage/basics/supported_k8s_versions.md [hunk 16]
```diff
@@ -12,7 +17,7 @@ The minimum version of a seed cluster that can be connected to Gardener is **`1.
 
 ## Shoot Clusters
 
-Gardener itself is capable of spinning up clusters with Kubernetes versions **`1.25`** up to **`1.30`**.
+Gardener itself is capable of spinning up clusters with Kubernetes versions **`1.25`** up to **`1.29`**.
```
Comment (ialidzhikov) [inline docs/usage/basics/supported_k8s_versions.md]: Revert this change, this is issue during the rebase, gardener supports K8s 1.30.

File: docs/usage/basics/_index.md [hunk 17]
```diff
@@ -0,0 +1,5 @@
+---
```
Comment (ialidzhikov) [inline docs/usage/basics/_index.md]: Use imperative for the commit messages. Use `Add basics section` instead of `Adding basics section` or `Added basic section`.

File: docs/usage/basics/_index.md [hunk 18]
```diff
@@ -0,0 +1,5 @@
...
+title: Basics
+description: No prior Gardener knowledge is required
```
Comment (ialidzhikov) [inline docs/usage/basics/_index.md]: Does it make sense to add more detailed description what is covered in this Basics section? Something like in this section you will learn how to create Shoot, access it, etc.

File: docs/usage/basics/basic-configuration.md [hunk 19]
```diff
@@ -0,0 +1,6 @@
+---
+title: Basic Shoot configurations
+weight: 6
+---
+
+The [gardener dashboard](https://github.com/gardener/dashboard/blob/master/README.md) provides a simple way for creating and configuring a basic shoot cluster. You can see how changing the shoot configuration through the overview section reflects on the shoot yaml resource. When talking about clusters in the documentation the yaml structure will be mostly refered and therefore a basic understanding of it's structure will be highly usefull.
```
Comment (ialidzhikov) [inline docs/usage/basics/basic-configuration.md]: You can run a spell-checker locally for the newly introduced content.

```suggestion
The [gardener dashboard](https://github.com/gardener/dashboard/blob/master/README.md) provides a simple way for creating and configuring a basic shoot cluster. You can see how changing the shoot configuration through the overview section reflects on the shoot YAML resource. When talking about clusters in the documentation the YAML structure will be mostly referred and therefore a basic understanding of it's structure will be highly useful.
```

Comment (ialidzhikov) [inline docs/usage/basics/basic-configuration.md]: I recall raising the same question already many times to @n-boshnakov . "shoot" vs "Shoot" in the docs? "gardener dashboard" vs "Gardener dashboard"? Right now Shoot is with upper-case in the titles but lower-case in this docs.

File: docs/usage/basics/creating-shoot.md [hunk 20]
```diff
@@ -0,0 +1,6 @@
+---
+title: Creating a Shoot cluster
+weight: 4
+---
+
+You can check the tutorial section for the different providers in [Infrastructure Extensions](/docs/extensions/infrastructure-extensions/)
```
Comment (ialidzhikov) [inline docs/usage/basics/creating-shoot.md]:
```suggestion
You can check the tutorial section for the different providers in [Infrastructure Extensions](/docs/extensions/infrastructure-extensions/).
```

File: docs/usage/basics/projects.md [hunk 21]
```diff
@@ -1,8 +1,11 @@
-# Projects
+---
+weight: 2
+description: Project operations, roles and what they allow. Explaining the Four-Eyes-Principle for resource deletion
```
Comment (ialidzhikov) [inline docs/usage/basics/projects.md]:
```suggestion
description: Project operations and roles. Four-Eyes-Principle for resource deletion
```

Comment (ialidzhikov) [inline docs/usage/basics/projects.md]: It is not really easy to review this PR as changes done in this commit for example are not final. Later on things are restructured again. It would be easier to review if commits contained the final changes.
As you have the changes done locally - does it make sense to split the changes to meaningful commits that can be reviewed separately?

File: docs/usage/basics/shoot_access.md [hunk 22]
```diff
@@ -1,3 +1,8 @@
+---
+title: Accessing Shoot Clusters
```
Comment (ialidzhikov) [inline docs/usage/basics/shoot_access.md]: It is not clear at all for me when you decide to set a title or not? Was it documented somewhere? I recall that hugo (?) generates a title based on the title in the markdown. However things here seems to be straightforward so why we add title here but for example we don't add title to the `service-account-manager.md` ?

File: docs/usage/shoot_settings/default_seccomp_profile.md [hunk 23]
```diff
@@ -0,0 +1,30 @@
+---
+title: Default Seccomp Profile
+description: Enable the use of `RuntimeDefault` as the default seccomp profile through `spec.kubernetes.kubelet.seccompDefault`
```
Comment (ialidzhikov) [inline docs/usage/shoot_settings/default_seccomp_profile.md:3]: I would avoid adding the field as kubelet settings can be provided globally (https://github.com/gardener/gardener/blob/98fb871eb6045f69715d9f8b39dd84f9637f1fa0/example/90-shoot.yaml#L237, `spec.kubernetes.kubelet`) and they can be also provided per worker pool (https://github.com/gardener/gardener/blob/98fb871eb6045f69715d9f8b39dd84f9637f1fa0/example/90-shoot.yaml#L79, `spec.provider.workers[].kubernetes.kubelet`).

File: docs/usage/shoot_settings/default_seccomp_profile.md [hunk 24]
```diff
@@ -0,0 +1,30 @@
...
+---
+
+# Default Seccomp Profile and Configuration 
+
+This is a short guide describing how to enable the defaulting of seccomp profiles for Gardener managed workloads in the seed. Running pods in `Unconfined` (seccomp disabled) mode is undesirable since this is the least restrictive profile. Also, mind that any privileged container will always run as `Unconfined`. More information about seccomp can be found in this [Kubernetes tutorial](https://kubernetes.io/docs/tutorials/security/seccomp/).
```
Comment (ialidzhikov) [inline docs/usage/shoot_settings/default_seccomp_profile.md:9]:
```suggestion
This is a short guide describing how to enable the defaulting of seccomp profiles for Gardener managed workloads for a Shoot cluster. Running pods in `Unconfined` (seccomp disabled) mode is undesirable since this is the least restrictive profile. Also, mind that any privileged container will always run as `Unconfined`. More information about seccomp can be found in this [Kubernetes tutorial](https://kubernetes.io/docs/tutorials/security/seccomp/).
```

File: docs/usage/shoot_settings/default_seccomp_profile.md [hunk 25]
```diff
@@ -0,0 +1,30 @@
...
+
+## Setting the Seccomp Profile to RuntimeDefault for Shoot Clusters
+
+You can enable the use of `RuntimeDefault` as the default seccomp profile for all workloads. If enabled, the kubelet will use the `RuntimeDefault` seccomp profile by default, which is defined by the container runtime, instead of using the `Unconfined` mode. More information for this feature can be found in the [Kubernetes documentation](https://kubernetes.io/docs/tutorials/security/seccomp/#enable-the-use-of-runtimedefault-as-the-default-seccomp-profile-for-all-workloads).
+
+To use seccomp profile defaulting, you must run the kubelet with the `SeccompDefault` feature gate enabled (this is the default).
```
Comment (ialidzhikov) [inline docs/usage/shoot_settings/default_seccomp_profile.md:15]: `(this is the default)` - this is not a good explanation. Enabled by default since which K8s version? What is the feature gate state across the different K8s versions?

File: docs/usage/shoot_settings/pod-security.md [hunk 26]
```diff
@@ -1,3 +1,7 @@
+---
+title: Admission Configuration for the `PodSecurity` Admission Plugin
+description: Adding custom configuration for the `PodSecurity` plugin in `.spec.kubernetes.kubeAPIServer.admissionPlugins`
+---
```
Comment (ialidzhikov) [inline docs/usage/shoot_settings/pod-security.md:5]: You could add an empty line:
```suggestion
---

```

File: docs/usage/shoot_settings/shoot_autoscaling.md [hunk 27]
```diff
@@ -23,7 +28,7 @@ There are [general options for `cluster-autoscaler`](../api-reference/core.md#co
...
```
Comment (ialidzhikov) [inline docs/usage/shoot_settings/shoot_autoscaling.md]:
```suggestion
The reason is that it was only introduced lately, and some end-users might have already deployed their own VPA into their clusters, i.e., enabling it by default would interfere with such custom deployments and lead to issues, eventually. Additionally, if the `ShootVPAEnabledByDefault` admission plugin is enabled in the gardener-apiserver, then the `.spec.kubernetes.verticalPodAutoscaler.enabled` field will be defaulted to true.
```

Add also a link to the admission plugin docs in `apiserver-admission-plugins.md`.

File: docs/usage/shoot_settings/shoot_hibernate.md [hunk 28]
```diff
@@ -1,10 +1,7 @@
 ---
-title: Hibernate a Cluster
-level: beginner
-category: Operation
-scope: operator
-tags: ["task"]
-publishdate: 2020-11-19
+title: Shoot Hybernation
```
Comment (ialidzhikov) [inline docs/usage/shoot_settings/shoot_hibernate.md:2]:
```suggestion
title: Shoot Hibernation
```

File: docs/usage/shoot_settings/shoot_hibernate.md [hunk 29]
```diff
@@ -1,10 +1,7 @@
...
+description: What is hybernation? Manual hybernation/wake up and specifying a hybernation schedule
```
Comment (ialidzhikov) [inline docs/usage/shoot_settings/shoot_hibernate.md:3]:
```suggestion
description: What is hibernation? Manual hibernation/wake up and specifying a hibernation schedule
```

File: docs/usage/shoot_settings/shoot_hibernate.md [hunk 30]
```diff
@@ -1,10 +1,7 @@
...
+weight: 2
 ---
```
Comment (ialidzhikov) [inline docs/usage/shoot_settings/shoot_hibernate.md:5]: Add empty line:

```suggestion
---

```

File: docs/usage/shoot_settings/shoot_hibernate.md [hunk 31]
```diff
@@ -30,22 +19,22 @@ When a cluster is hibernated, Gardener scales down the worker nodes and the clus
...
+### What Isn’t Affected by the Hibernation?
 
 To scale up everything where it was before hibernation, Gardener doesn’t delete state-related information, that is, information stored in persistent volumes. The cluster state as persistent in `etcd` is also preserved.
 
 ## Hibernate Your Cluster Manually
 
 The `.spec.hibernation.enabled` field specifies whether the cluster needs to be hibernated or not. If the field is set to `true`, the cluster's desired state is to be hibernated. If it is set to `false` or not specified at all, the cluster's desired state is to be awakened.
 
-To hibernate your cluster, you can run the following `kubectl` command:
+To hibernate your cluster, you can do it from the dashboard or run the following `kubectl` command:
```
Comment (ialidzhikov) [inline docs/usage/shoot_settings/shoot_hibernate.md:30]: Which dashboard? The K8s dashboard or the Gardener dashboard?

```suggestion
To hibernate your cluster, you can do it from the Gardener dashboard or run the following `kubectl` command:
```

File: docs/usage/shoot_settings/shoot_hibernate.md [hunk 32]
```diff
@@ -30,22 +19,22 @@ When a cluster is hibernated, Gardener scales down the worker nodes and the clus
...
 ```
 $ kubectl patch shoot -n $NAMESPACE $SHOOT_NAME -p '{"spec":{"hibernation":{"enabled": true}}}'
 ```
 
 ## Wake Up Your Cluster Manually
 
-To wake up your cluster, you can run the following `kubectl` command:
+To wake up your cluster, you can do it from the dashboard or run the following `kubectl` command:
```
Comment (ialidzhikov) [inline docs/usage/shoot_settings/shoot_hibernate.md:37]:
```suggestion
To wake up your cluster, you can do it from the Gardener dashboard or run the following `kubectl` command:
```

File: docs/usage/shoot_settings/shoot_high_availability.md [hunk 33]
```diff
@@ -1,3 +1,7 @@
+---
+weight: 7
+description: Node and Zone Failure Tolerance and possible mitigations for Zone Outages
+---
```
Comment (ialidzhikov) [inline docs/usage/shoot_settings/shoot_high_availability.md:4]: "possible mitigations for Zone Outages": we have to reword this.

Comment (ialidzhikov) [inline docs/usage/shoot_settings/shoot_high_availability.md:4]:
```suggestion
---

```

**Review by ialidzhikov — COMMENTED** [review]
Hi @Kostov6 

I give up on the review on this PR on the 2nd commit as I think the PR does too much things and it is hard to track everything as a reviewer.
To proceed forward I would suggest you to split the changes from this PR to dedicated PRs to make possible reviewing the changes easily:
- One PR that updates the titles and adds descriptions
- One or several PRs that touch the content itself - splitting docs for personas, other docs changes
- One or several PRs that introduce the content structure changes - adding the new sections, moving the docs to these sections.

I think in this way we can get the changes reviewed in a faster and more reliable way. :)

File: docs/usage/basics/shoot_access.md (hunk 22 above)
Comment (ialidzhikov) [reply docs/usage/basics/shoot_access.md]: I don't recall whether we have a developer guide for docs: for example, use relative links, etc. If we have such guide, we can enhance it with the new things from this PR: for example, add description to your docs to improve the search quality in the website.

**Comment by ialidzhikov** [comment]
I am closing this PR as we agreed with @Kostov6 to have the changes split in dedicated PRs.

/close
